### PR TITLE
Fix EFFICIENCY calculation

### PR DESCRIPTION
**Pull request by mladenbanovic**
## Proposed Changes
This fix resolves a problem related to mismatched gradients between the forward and reverse mode of AD when considering EFFICIENCY as the objective function.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X ] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [ ] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: SU2_CFD/include/solvers/CFVMFlowSolverBase.inl [hunk 1]
```diff
@@ -2337,7 +2337,7 @@ void CFVMFlowSolverBase<V, FlowRegime>::Friction_Forces(const CGeometry* geometr
           SurfaceViscCoeff.CL[iMarker_Monitoring] += ViscCoeff.CL[iMarker];
           SurfaceViscCoeff.CD[iMarker_Monitoring] += ViscCoeff.CD[iMarker];
           SurfaceViscCoeff.CSF[iMarker_Monitoring] += ViscCoeff.CSF[iMarker];
-          SurfaceViscCoeff.CEff[iMarker_Monitoring] += ViscCoeff.CEff[iMarker];
+          SurfaceViscCoeff.CEff[iMarker_Monitoring] = ViscCoeff.CEff[iMarker];
```
Comment (mladenbanovic) [inline SU2_CFD/include/solvers/CFVMFlowSolverBase.inl:2340]: Not sure about this line modification, but I replaced += with = to keep the code with respect to Efficiency consistent.

Comment (pcarruscag) [reply SU2_CFD/include/solvers/CFVMFlowSolverBase.inl:2340]: I think consistency wise you would divide SurfaceViscCoeff.CL by CD like in the other methods.
Everything else looks good.

**Review by pcarruscag — APPROVED** [review]
LGTM :+1:

**Comment by koodlyakshay** [comment]
Not directly related to this PR, but I noticed an potential bug in the Friction Forces routine, line 2258. The if condition checks 
    
    if (iDim == 3)
instead of 

    if (nDim == 3)
Could you fix this one too?

**Comment by pcarruscag** [comment]
Good catch, same bug exists in the Momentum forces routine.

**Comment by pcarruscag** [comment]
I will fix the iDim==3 issues in #1022

**Pull request merged**